### ISSUE TITLE
Cause of "No documentation available"?

### ISSUE THREAD

**Issue by jberdine** [the-lambda-church/merlin#558]
From emacs, merlin-document successfully finds documentation for e.g. functions defined in the standard library.  For functions defined in the current project, I seem to always get "No documentation available".  This happens for identifiers which merlin-locate can correctly find, and which have documentation comments.  The .merlin file configures S B and CMT directories.  I have checked that the .cmt files are actually produced.

Any hints on how to debug this / what I have missed?

**Comment by trefis** [comment]
Are you using camlp4 for your project?

**Comment by jberdine** [comment]
> Are you using camlp4 for your project?

No, no camlp4.  The repo is https://github.com/facebook/infer/tree/master/infer if it helps.

**Comment by trefis** [comment]
And you are sure you are compiling with `-bin-annot`? (i.e. are there `.cmt` and `.cmti` files in your build directory?)

**Comment by jberdine** [comment]
On Tue, Aug 02 2016, Thomas Refis wrote:

> And you are sure you are compiling with -bin-annot? (i.e. are there .cmt and .cmti files in your build
> directory?)

Yes: https://github.com/facebook/infer/blob/master/infer/src/Makefile#L41

(And I also looked in the build directory.)

**Comment by trefis** [comment]
Hello,

Just to let you know: I haven't had the time to look at this yet, but I haven't forgotten about it. I should have the time to try and reproduce in the next few hopefully.
I'll keep you posted!

**Comment by let-def** [comment]
Infer is compiled by reason, I don't know if doc comments are preserved, the protocol is still a bit too limited.

**Comment by rgrinberg** [comment]
@Chris00 isn't go to definition on the module itself good enough?

**Comment by Chris00** [comment]
Well, `merlin-locate` sends to the `.ml` file from which one has to switch to the `.mli` file.  It would be convenient to directly go to the location in the `.mli` file so as to explore the surrounding of the value exported.  In Emacs, `merlin-document` with a a prefix (<kbd>C-u</kbd>) could do that.

**Comment by rgrinberg** [comment]
I see. So i think this will be fixed with lsp's `document/goToDeclaration` call.
